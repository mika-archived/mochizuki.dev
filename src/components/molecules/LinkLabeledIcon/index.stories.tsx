import React from "react";
import styled from "styled-components";
import { storiesOf } from "@storybook/react";

import LinkLabeledIcon from ".";

const SmallLabeledIcon = styled(LinkLabeledIcon)`
  font-size: 14px;
  color: #888;
`;

const LimitedWidth = styled.div`
  width: 150px;
`;

storiesOf("molecules/LinkLabeledIcon", module).add("default", () => (
  <>
    <LinkLabeledIcon href="https://github.com/mika-sandbox/dotnet-window-capture" icon="book" prefix="light" label="mika-sandbox/dotnet-window-capture" />
    <LimitedWidth>
      <LinkLabeledIcon href="https://github.com" icon="book" prefix="light" label="mika-sandbox/very-long-long-repository-name" />
    </LimitedWidth>
    <SmallLabeledIcon href="https://twitter.com" icon="star" prefix="solid" label="12" />
  </>
));
