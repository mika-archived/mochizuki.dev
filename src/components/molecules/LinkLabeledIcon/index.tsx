import React from "react";
import styled from "styled-components";

import FontAwesome from "../../atoms/FontAwesome";
import Link from "../../atoms/Link";
import { Paragraph } from "../../atoms/Typography";

import { ExtractProps } from "../../../type-helper";

type Props = {
  className?: string;
  href: string;
  icon: string;
  label: string;
  prefix: ExtractProps<typeof FontAwesome>["prefix"];
};

const Anchor = styled(Link)`
  color: inherit;
  text-decoration: none;
`;

const FlexBox = styled.div`
  display: flex;
  flex-flow: row nowrap;
`;

const Label = styled(Paragraph)`
  margin-left: 4px;
  overflow: hidden;
  line-height: 1.2;
  text-overflow: ellipsis;
  white-space: nowrap;
`;

const LinkLabeledIcon: React.FC<Props> = ({ className, href, icon, label, prefix }) => {
  return (
    <FlexBox className={className}>
      <FontAwesome fixed icon={icon} prefix={prefix} />
      <Label>
        <Anchor href={href}>{label}</Anchor>
      </Label>
    </FlexBox>
  );
};

export default LinkLabeledIcon;
