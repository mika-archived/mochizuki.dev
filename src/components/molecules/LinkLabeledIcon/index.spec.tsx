import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";

import React from "react";
import { cleanup, render } from "@testing-library/react";

import LinkLabeledIcon from ".";

afterEach(cleanup);

test("does not have changes from snapshot", () => {
  const { asFragment } = render(<LinkLabeledIcon href="https://github.com" icon="book" prefix="light" label="mika-sandbox/repository-name" />);

  expect(asFragment()).toMatchSnapshot();
});
