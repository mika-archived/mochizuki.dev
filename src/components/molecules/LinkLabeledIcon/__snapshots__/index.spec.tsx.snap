// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not have changes from snapshot 1`] = `
<DocumentFragment>
  .c2 {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
}

.c1 {
  margin: 0;
  font-family: "游ゴシック体","Yu Gothic",YuGothic,"ヒラギノ角ゴシック Pro","Hiragino Kaku Gothic Pro",メイリオ,Meiryo,sans-serif;
  margin-left: 4px;
  overflow: hidden;
  line-height: 1.2;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<div
    class="c0"
  >
    <i
      class="fal fa-book fa-fw"
    />
    <p
      class="sc-AxjAm c1"
    >
      <a
        class="c2"
        href="https://github.com"
        rel="noreferrer noopener"
        target="_blank"
      >
        mika-sandbox/repository-name
      </a>
    </p>
  </div>
</DocumentFragment>
`;
